binderhub/base.py

```diff
@@ -2,9 +2,45 @@
 
 from http.client import responses
 from tornado import web
+from jupyterhub.services.auth import HubAuthenticated, HubOAuth
+import functools
+from urllib.parse import urlencode
 
 
-class BaseHandler(web.RequestHandler):
+def authenticated(method):
+    """Copied from tornado.web.authenticated and `auth_enabled` condition is added.
+    If authentication in not enabled, this decorator doesn't do anything.
+    """
+    @functools.wraps(method)
+    def wrapper(self, *args, **kwargs):
```

Comment:
Rather than reimplementing `@web.authenticated`, we can define a `.get_current_user` that returns `'anonymous'` if auth is not enabled.